### ISSUE TITLE
Tests fail with Sphinx 1.8.2

### ISSUE THREAD

**Issue by mitya57** [sphinx-doc/sphinxcontrib-websupport#18]
The first test passes, but the second one fails with:
```python
Traceback (most recent call last):
  File "/home/dmitry/upstream/sphinxcontrib-websupport/tests/test_websupport.py", line 63, in test_build
    support.build()
  File "/usr/lib/python3/dist-packages/sphinxcontrib/websupport/core.py", line 137, in build
    app.build()
  File "/home/dmitry/upstream/sphinx/sphinx/application.py", line 341, in build
    self.builder.build_update()
  File "/home/dmitry/upstream/sphinx/sphinx/builders/__init__.py", line 347, in build_update
    len(to_build))
  File "/home/dmitry/upstream/sphinx/sphinx/builders/__init__.py", line 412, in build
    self.write(docnames, list(updated_docnames), method)
  File "/home/dmitry/upstream/sphinx/sphinx/builders/__init__.py", line 593, in write
    self._write_serial(sorted(docnames))
  File "/home/dmitry/upstream/sphinx/sphinx/builders/__init__.py", line 604, in _write_serial
    self.write_doc(docname, doctree)
  File "/home/dmitry/upstream/sphinxcontrib-websupport/sphinxcontrib/websupport/builder.py", line 75, in write_doc
    self.docwriter.write(doctree, destination)
  File "/usr/lib/python3/dist-packages/docutils/writers/__init__.py", line 80, in write
    self.translate()
  File "/home/dmitry/upstream/sphinx/sphinx/writers/html.py", line 58, in translate
    self.document.walkabout(visitor)
  File "/usr/lib/python3/dist-packages/docutils/nodes.py", line 174, in walkabout
    if child.walkabout(visitor):
  File "/usr/lib/python3/dist-packages/docutils/nodes.py", line 174, in walkabout
    if child.walkabout(visitor):
  File "/usr/lib/python3/dist-packages/docutils/nodes.py", line 174, in walkabout
    if child.walkabout(visitor):
  [Previous line repeated 3 more times]
  File "/usr/lib/python3/dist-packages/docutils/nodes.py", line 166, in walkabout
    visitor.dispatch_visit(self)
  File "/home/dmitry/upstream/sphinxcontrib-websupport/sphinxcontrib/websupport/writer.py", line 27, in dispatch_visit
    self.handle_visit_commentable(node)
  File "/home/dmitry/upstream/sphinxcontrib-websupport/sphinxcontrib/websupport/writer.py", line 34, in handle_visit_commentable
    self.add_db_node(node)
  File "/home/dmitry/upstream/sphinxcontrib-websupport/sphinxcontrib/websupport/writer.py", line 46, in add_db_node
    source=node.rawsource or node.astext())
  File "/usr/lib/python3/dist-packages/sphinxcontrib/websupport/storage/sqlalchemystorage.py", line 53, in add_node
    self.build_session.flush()
  File "/usr/lib/python3/dist-packages/sqlalchemy/orm/session.py", line 2254, in flush
    self._flush(objects)
  File "/usr/lib/python3/dist-packages/sqlalchemy/orm/session.py", line 2380, in _flush
    transaction.rollback(_capture_exception=True)
  File "/usr/lib/python3/dist-packages/sqlalchemy/util/langhelpers.py", line 66, in __exit__
    compat.reraise(exc_type, exc_value, exc_tb)
  File "/usr/lib/python3/dist-packages/sqlalchemy/util/compat.py", line 187, in reraise
    raise value
  File "/usr/lib/python3/dist-packages/sqlalchemy/orm/session.py", line 2344, in _flush
    flush_context.execute()
  File "/usr/lib/python3/dist-packages/sqlalchemy/orm/unitofwork.py", line 391, in execute
    rec.execute(self)
  File "/usr/lib/python3/dist-packages/sqlalchemy/orm/unitofwork.py", line 556, in execute
    uow
  File "/usr/lib/python3/dist-packages/sqlalchemy/orm/persistence.py", line 181, in save_obj
    mapper, table, insert)
  File "/usr/lib/python3/dist-packages/sqlalchemy/orm/persistence.py", line 830, in _emit_insert_statements
    execute(statement, multiparams)
  File "/usr/lib/python3/dist-packages/sqlalchemy/engine/base.py", line 948, in execute
    return meth(self, multiparams, params)
  File "/usr/lib/python3/dist-packages/sqlalchemy/sql/elements.py", line 269, in _execute_on_connection
    return connection._execute_clauseelement(self, multiparams, params)
  File "/usr/lib/python3/dist-packages/sqlalchemy/engine/base.py", line 1060, in _execute_clauseelement
    compiled_sql, distilled_params
  File "/usr/lib/python3/dist-packages/sqlalchemy/engine/base.py", line 1200, in _execute_context
    context)
  File "/usr/lib/python3/dist-packages/sqlalchemy/engine/base.py", line 1413, in _handle_dbapi_exception
    exc_info
  File "/usr/lib/python3/dist-packages/sqlalchemy/util/compat.py", line 203, in raise_from_cause
    reraise(type(exception), exception, tb=exc_tb, cause=cause)
  File "/usr/lib/python3/dist-packages/sqlalchemy/util/compat.py", line 186, in reraise
    raise value.with_traceback(tb)
  File "/usr/lib/python3/dist-packages/sqlalchemy/engine/base.py", line 1193, in _execute_context
    context)
  File "/usr/lib/python3/dist-packages/sqlalchemy/engine/default.py", line 508, in do_execute
    cursor.execute(statement, parameters)
sqlalchemy.exc.IntegrityError: (sqlite3.IntegrityError) UNIQUE constraint failed: sphinx_nodes.id [SQL: 'INSERT INTO sphinx_nodes (id, document, source) VALUES (?, ?, ?)'] [parameters: ('31a6bbc649b040b3889bf5e4a4ae6d80', 'lists', 'description')] (Background on this error at: http://sqlalche.me/e/gkpj)
```

**Comment by mitya57** [comment]
`git bisect` tells me that it broke after sphinx-doc/sphinx@6b14cd3bd62cec3d653965cfa7384e98c2c93bf7.

**Comment by mitya57** [comment]
@tk0miya Can you please take a look at this?

**Comment by tk0miya** [comment]
I don't understand what is happening.
@shimizukawa Could you help us please?

**Comment by tk0miya** [comment]
Wait. sorry, I just overlooked #20.

**Comment by tk0miya** [comment]
Fixed by #20.
@mitya57 Great work!
